classes/local/observer/olduser.php

```diff
@@ -0,0 +1,74 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * @package tool
+ * @subpackage mergeusers
+ * @author Jordi Pujol-Ahulló <jordi.pujol@urv.cat>
+ * @author John Hoopes <hoopes@wisc.edu>, University of Wisconsin - Madison
+ * @copyright 2013 Servei de Recursos Educatius (http://www.sre.urv.cat)
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_mergeusers\local\observer;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+require_once $CFG->libdir . '/gdlib.php';
+
+class olduser {
+
+    /**
+     * Suspend the old user, by suspending its account, and updating the profile picture
+     * to a generic one
+     *
+     * @param object $event stdClass with all event data.
+     */
+    public static function old_user_suspend($event) {
```

Comment:
Added param and return types. You may want to update your readme to indicate that the min PHP version is now 7.1 as we are using return types. Moodle 3.4 supports php7.0, php7.1, php7.2

Reply:
It would make more sense to simply change the minimal supported Moodle version to Moodle 3.9 (oldest supported Moodle version), this way we have no extra limitation on PHP version.

Reply:
According to [Moodle release notes](https://docs.moodle.org/dev/Moodle_3.9_release_notes#Server_requirements), minimal PHP version is 7.2.0. So doing like @ndunand said, it would be fine.

So next plugin release would support only M3.9 and forward. That's easy.